src/sentry/api/endpoints/organization_discover_query.py

```diff
@@ -303,7 +308,8 @@ def post(self, request, organization):
 
         has_aggregations = len(serialized.get('aggregations')) > 0
 
-        selected_columns = [] if has_aggregations else serialized.get('fields')
+        selected_columns = serialized.get(
+            'condition_fields') or [] if has_aggregations else serialized.get('fields')
```

Comment:
Can we merge condition_fields and fields if both are provided rather than ignore 1 over the other